test-browser/tests/plugin.js

```diff
@@ -0,0 +1,93 @@
+  },
+
+  'Should Search for plugins': function (browser) {
+    browser.waitForElementVisible('div[data-id="pluginManagerComponentPluginManager"]')
+     .click('input[data-id="pluginManagerComponentSearchInput"]')
+     .keys('debugger')
+     .waitForElementVisible('button[data-id="pluginManagerComponentActivateButtondebugger"]')
+     .clearValue('input[data-id="pluginManagerComponentSearchInput"]')
+     .click('input[data-id="pluginManagerComponentSearchInput"]')
+     .keys('Deploy & run transactions')
+     .waitForElementVisible('button[data-id="pluginManagerComponentActivateButtonudapp"]')
+     .clearValue('input[data-id="pluginManagerComponentSearchInput"]')
+     .click('input[data-id="pluginManagerComponentSearchInput"]')
+     .keys('ZoKrates')
+     .waitForElementVisible('button[data-id="pluginManagerComponentActivateButtonZoKrates"]')
+     .clearValue('input[data-id="pluginManagerComponentSearchInput"]')
+     .click('input[data-id="pluginManagerComponentSearchInput"]')
+     .keys(browser.Keys.ENTER)
+  },
+
+  'Should activate plugins': function (browser) {
+    browser.waitForElementVisible('div[data-id="pluginManagerComponentPluginManager"]')
+    .click('div[data-id="pluginManagerComponentPluginManager"]')
+    .scrollAndClick('button[data-id="pluginManagerComponentActivateButtondebugger"]')
+    .waitForElementVisible('button[data-id="pluginManagerComponentDeactivateButtondebugger"]')
+    .scrollAndClick('button[data-id="pluginManagerComponentActivateButtonudapp"]')
+    .waitForElementVisible('button[data-id="pluginManagerComponentDeactivateButtonudapp"]')
+    .scrollAndClick('button[data-id="pluginManagerComponentActivateButtonZoKrates"]')
+    .waitForElementVisible('button[data-id="pluginManagerComponentDeactivateButtonZoKrates"]')
+  },
+
+  'Should deactivate plugins': function (browser) {
+    browser.waitForElementVisible('div[data-id="pluginManagerComponentPluginManager"]')
+    .click('div[data-id="pluginManagerComponentPluginManager"]')
+    .scrollAndClick('button[data-id="pluginManagerComponentDeactivateButtondebugger"]')
+    .waitForElementVisible('button[data-id="pluginManagerComponentActivateButtondebugger"]')
+    .scrollAndClick('button[data-id="pluginManagerComponentDeactivateButtonudapp"]')
+    .waitForElementVisible('button[data-id="pluginManagerComponentActivateButtonudapp"]')
+  },
+
+  'Grant plugin permission (ZOKRATES)': function (browser) {
+    browser.waitForElementVisible('div[data-id="pluginManagerComponentPluginManager"]')
+    .click('button[data-id="pluginManagerSettingsButton"]')
+    .waitForElementVisible('form[data-id="pluginManagerSettingsPermissionForm"]')
+    .assert.containsText('form[data-id="pluginManagerSettingsPermissionForm"]', 'No Permission requested yet')
+    .modalFooterOKClick()
+    .click('div[data-id="verticalIconsFileExplorerIcons"]')
+    .switchFile('browser/3_Ballot.sol')
+    .click('div[plugin="ZoKrates"]')
+    .pause(5000)
+    .frame(0)
```

Comment:
what is doing `frame` ?

Reply:
`.frame(0)` allows switching into the first `<iframe>` indexed at 0 (plugin manager's iframe) and selecting DOM elements. I added this because i needed to select the compile button in ZoKrates which is inside an iframe.